confidant/authnz/userauth.py

```diff
@@ -0,0 +1,661 @@
+import abc
+import logging
+import urlparse
+
+import yaml
+
+import flask
+from flask import request, session
+from flask import abort, jsonify, redirect
+
+# google auth imports
+from authomatic import Authomatic
+from authomatic.providers import oauth2
+from authomatic.adapters import WerkzeugAdapter
+
+# saml auth imports
+from onelogin.saml2.auth import OneLogin_Saml2_Auth
+from confidant.lib import cryptolib
+from confidant.utils.misc import dict_deep_update
+
+from confidant.app import app
+
+from . import errors
```

Comment:
When we released we killed off all relative imports. I'm not strongly opposed to them, but would be nice if this could be an explicit import from confidant.authnz